terraform/helm/genesis/templates/genesis.yaml

```diff
@@ -0,0 +1,160 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "aptos-genesis.fullname" . }}-layout-e{{ .Values.chain.era }}
+  labels:
+    {{- include "aptos-genesis.labels" . | nindent 4 }}
+data:
+  layout.yaml: |-
+    ---
+    root_key: '0x5243ca72b0766d9e9cbf2debf6153443b01a1e0e6d086c7ea206eaf6f8043956'
+    users:
+    {{- range $user := .Values.genesis.users }}
+    - {{ $user.name | squote }}
+    {{- end }}
+    chain_id: {{ .Values.chain.chain_id | int }}
+
+---
+
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: {{ include "aptos-genesis.fullname" . }}
+  labels:
+    {{- include "aptos-genesis.labels" . | nindent 4 }}
+rules:
+- apiGroups: [""]
+  verbs: ["list", "create", "delete"]
+  resources: ["configmaps", "secrets", "persistentvolumeclaims"]
+
+---
+
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: {{ include "aptos-genesis.fullname" . }}
+  labels:
+    {{- include "aptos-genesis.labels" . | nindent 4 }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: {{ include "aptos-genesis.fullname" . }}
+subjects:
+- kind: ServiceAccount
+  name: {{ include "aptos-genesis.serviceAccountName" . }}
+
+---
+
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ include "aptos-genesis.fullname" . }}-e{{ .Values.chain.era }}
+  labels:
+    {{- include "aptos-genesis.labels" . | nindent 4 }}
+    app.kubernetes.io/name: genesis
+spec:
+  template:
+    metadata:
+      labels:
+        {{- include "aptos-genesis.selectorLabels" . | nindent 8 }}
+        app.kubernetes.io/name: genesis
+      annotations:
+        seccomp.security.alpha.kubernetes.io/pod: runtime/default
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: genesis
+        image: {{ .Values.genesis.image.repo }}:{{ .Values.genesis.image.tag | default .Values.imageTag }}
+        imagePullPolicy: {{ .Values.genesis.image.pullPolicy }}
+        workingDir: /tmp
+        command:
+        - sh
+        - -c
+        - |-
+          set -ex
+          WORKSPACE=/tmp
```

Comment:
I think it'd be better to put this into an untemplated external script and then mount the script via configmap.
Having such a complex script inline in a yaml template and having the script itself being templates, is a maintenance / testing burden.

Reply:
here's an example of how to do that: https://suraj.io/post/use-configmap-for-scripts/

Reply:
Yea I thought of doing that too, but it got too complex to make untemplated, since we rely on `.genesis.users`, which is a list of maps

Reply:
I wonder if everything can be derived based on their naming structure. The only input you need is number of validator nodes, and then you can assume validator/fullnode LB is just the format we set. Since genesis will only be used along with the validator helm chart for now.

Reply:
I was thinking it might be nice to be able to name validators rather than have them implicitly named by their index. I need to give that a bit more thinking though, but for now, I can move to a separate untemplated script for readability